src/core/framegraphes/forwardrenderer.cpp

```diff
@@ -339,7 +339,7 @@ ForwardRenderer::ForwardRenderer(Qt3DCore::QNode *parent)
 ForwardRenderer::~ForwardRenderer()
 {
     // unparent the effect subtrees or they'll be deleted twice
-    for (auto framegraph : qAsConst(m_effectFGSubtrees))
+    for (auto &framegraph : qAsConst(m_effectFGSubtrees))
```

Comment:
const auto & ?

Reply:
(though if it's only a pointer I wonder what is being gained)

Reply:
it's a `QSharedPointer` so we do save on a ref/unref. Can't be const since `QNode::setParent` is not const

Reply:
ah, indeed